lte/gateway/c/session_manager/UEShard.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <map>
+
+#include "UEShard.h"
+#define MAX_SHARD_SIZE 100
+
+namespace magma {
+
+UEShard::UEShard() {
+  number_of_shards = 0;
+}
+
+// returns shard_id associated with UE added
+int UEShard::add_ue(std::string imsi) {
+  int shard_id;
+  if (shards.size() == 0) {
+    shard_id = 1;
```

Comment:
Unless I recommend to start with 0. It will make your life easier

Reply:
Also, maybe explore the option of using an array/vector and initialize each shard, instead of use a dictionary. You can save the need for a key.

Reply:
Array/vector definitely makes sense, the hashmap isn't necessary if the ids for shards are as such. Good point.